### PR TITLE
MM-13498: fix subpath assets rewrite

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
The subpath rewrite that occurs on server start is tightly coupled to the definition of `root.html`, but the assumption was that this won't change often. It /did/ change, and appears to have occurred alongside the Webpack upgrade. Add an error log should this happen in the future.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13498

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: utils/subpath.go [hunk 1]
```diff
@@ -69,9 +69,13 @@ func UpdateAssetsSubpath(subpath string) error {
 	script := fmt.Sprintf("window.publicPath='%s'", newPath)
 	scriptHash := sha256.Sum256([]byte(script))
 
-	reCSP := regexp.MustCompile(`<meta http-equiv=Content-Security-Policy content="script-src 'self' cdn.segment.com/analytics.js/ 'unsafe-eval'([^"]*)">`)
+	reCSP := regexp.MustCompile(`<meta http-equiv="Content-Security-Policy" content="script-src 'self' cdn.segment.com/analytics.js/ 'unsafe-eval'([^"]*)">`)
```
Comment (levb) [inline utils/subpath.go]: Assuming the previous implementation was meaningful, should the new quotes be optional? Or do we only apply `reCSP` to the inputs we control, below?

File: utils/subpath_test.go [hunk 2]
```diff
@@ -193,19 +215,21 @@ func TestGetSubpathFromConfig(t *testing.T) {
...
...reen{font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;padding-top:50px;max-width:750px;font-size:14px;color:#333;margin:auto;display:none;line-height:1.5}.error-screen h2{font-size:30px;font-weight:400;line-height:1.2}.error-screen ul{padding-left:15px;line-height:1.7;margin-top:0;margin-bottom:10px}.error-screen hr{color:#ddd;margin-top:20px;margin-bottom:20px;border:0;border-top:1px solid #eee}.error-screen-visible{display:block}</style> <link href="/static/main.364fd054d7a6d741efc6.css" rel="stylesheet"><script type="text/javascript" src="/static/main.e49599ac425584ffead5.js"></script></head> <body class=font--open_sans> <div id=root> <div class=error-screen> <h2>Cannot connect to Mattermost</h2> <hr/> <p>We're having trouble connecting to Mattermost. If refreshing this page (Ctrl+R or Command+R) does not work, please verify that your computer is connected to the internet.</p> <br/> </div> <div class=loading-screen style=position:relative> <div class=loading__content> <div class="round round-1"></div> <div class="round round-2"></div> <div class="round round-3"></div> </div> </div> </div> <noscript> To use Mattermost, please enable JavaScript. </noscript> </body> </html>`
```
Comment (levb) [inline utils/subpath_test.go]: Should this be `"Content-Security-Policy"`?

File: utils/subpath.go (hunk 1 above)
Comment (lieut-data) [reply utils/subpath.go]: We only apply `reCSP` to inputs we control, specifically the `root.html` file that Webpack generates as part of the build process. Its previous output didn't escape the `http-equiv` attribute's value, but the new version does.

File: utils/subpath_test.go (hunk 2 above)
Comment (lieut-data) [reply utils/subpath_test.go]: In this case, no: I'm adding a test to verify that we fail and report an error if we can't find the string to replace. It's essentially the v5.5 version of `root.html`.

Comment (levb) [reply utils/subpath_test.go]: Ah! 👍

**Pull request merged**